update hardhat example with code and step by step instructions (#596)
onflow/docs: docs/evm/Build/Guides/Deploy Solidity Contract/deploy-solidity-contract-with-hardhat.md

    }
};

export default config;
```
For simplicity of this example, the account private key is used in hardhat.config.ts. Make sure to never check in a private key into your git repository. Better to use environment variables.

### Deploying HelloWorld Smart Contract

### HelloWorld Smart Contract
```Solidity
pragma solidity ^0.8.0;

contract HelloWorld {
    // Declare a public field of type string.
    string public greeting;

    // Constructor to initialize the greeting.
    // In Solidity, the constructor is defined with the "constructor" keyword.
    constructor() {
        greeting = "Hello, World!";
    }

    // Public function to change the greeting.
    // The "public" keyword makes the function accessible from outside the contract.
    function changeGreeting(string memory newGreeting) public {
        greeting = newGreeting;
    }

    // Public function that returns the greeting.
    // In Solidity, explicit return types are declared.
    function hello() public view returns (string memory) {
        return greeting;
    }
}

```

Deploying:
1. Create a file named HelloWorld.sol under contracts directory.
2. Add above HelloWorld.sol contract code to new file.

> 3. Create a deploy.ts file in `scripts` directory.

Reviewer: ```suggestion
3. Create a `deploy.ts` file in `scripts` directory.
```

4. Paste in the following TypeScript code.

```typeScript
import { ethers } from "hardhat";

async function main() {
    const HelloWorld = await ethers.getContractFactory("HelloWorld");
    console.log("Deploying HelloWorld...")
    const helloWorld = await HelloWorld.deploy();
    console.log("HelloWorld address:", helloWorld.address);
}

main()
    .then(() => process.exit(0))
    .catch((error) => {
        console.error(error);
        process.exit(1);
    });

```

> 5. Run ` npx hardhat run scripts/deploy.ts --network previewNet` in the project root.

Reviewer: ```suggestion
5. Run `npx hardhat run scripts/deploy.ts --network previewNet` in the project root.
```

Reviewer: should this be `previewnet` or `previewNet`

```TypeScript
import HelloWorldABI from "../contracts/HelloWorldABI.json"
import { ethers } from "hardhat";

async function main() {
    const contractAddress = "0x3Fe94f43Fb5CdB8268A801f274521a07F7b99dfb";

    const newGreeting = process.env.NEW_GREETING;
    if (!newGreeting) {
        console.error("Please set the NEW_GREETING environment variable.");
        process.exit(1);
    }

    // Signer to send the transaction (e.g., the first account from the hardhat node)
    const [signer] = await ethers.getSigners();

    // Contract instance with signer
    const helloWorldContract = new ethers.Contract(contractAddress, HelloWorldABI, signer);

    console.log("The greeting is:", await helloWorldContract.hello());

    // Create and send the transaction
    const tx = await helloWorldContract.changeGreeting(newGreeting);
    console.log("Transaction hash:", tx.hash);

    // Wait for the transaction to be mined
    await tx.wait().catch((error: Error) => {});
    console.log("Greeting updated successfully!");
    console.log("The greeting is:", await helloWorldContract.hello());
}

main().catch((error) => {
    console.error(error);
    process.exit(1);
});

```

> Script to get current greeting then update and lastly receive the new greeting. Steps:

Reviewer: ```suggestion
Next we'll add a script to update the greeting and log it. Here are the steps to follow:
```

1. Create new script `updateGreeting.ts` file in the `scripts` directory
2. Paste in the TypeScript above, Make sure to update the contract address with the one from deployment in earlier step. 
3. Call the new script, `NEW_GREETING='Howdy!' npx hardhat run ./scripts/updateGreeting.ts --network previewNet`
4. Output should be
```shell
❯ NEW_GREETING='Howdy!' npx hardhat run ./scripts/updateGreeting.ts --network previewNet
The greeting is: Hello, World!
Transaction hash: 0x03136298875d405e0814f54308390e73246e4e8b4502022c657f04f3985e0906
Greeting updated successfully!
The greeting is: Howdy!
```

> Small note, ignoring any errors that might occur. Verify that HelloWorld smart contract greeting has been updated correctly.

Reviewer: ```suggestion
Small note, ignoring any errors that might occur. Verify that the `HelloWorld` smart contract greeting has been updated correctly.
```

> 

Reviewer: Think we can remove all the below?

sidebar_position: 1
---

# Using Hardhat

Hardhat is a comprehensive Ethereum development tool that simplifies the process of deploying, testing, and debugging smart contracts. It's designed to offer developers a seamless experience when working with Solidity contracts.

## Prerequisites

### Software

Node v18+, can be downloaded [Here](https://nodejs.org/en/download)

Getting Started with Hardhat. [More Information](https://hardhat.org/tutorial/creating-a-new-hardhat-project). These directions demonstrate using npm commands to create a project and add its dependencies.

```shell
mkdir hardhat-example
cd hardhat-example

npm init

npm install --save-dev hardhat

npx hardhat init
```

### Fund Your Wallet

> Navigate to the [Crescendo PreviewNet Faucet](https://crescendo-faucet.onflow.org/fund-account). Paste in your wallet address to get receive $FLOW. Thi will allow your wallet address to deploy smart contracts. 

Reviewer: ```suggestion
Navigate to the [Crescendo PreviewNet Faucet](https://crescendo-faucet.onflow.org/fund-account). Paste in your wallet address to get receive $FLOW. This will allow your wallet address to deploy smart contracts and run transactions. 
```
